### PR TITLE
[-] Technical - Makes the JWT lighter and consistent across lianas

### PR DESCRIPTION
**Pull request by larcin**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [ ] Create automatic tests
- [x] No automatic tests failures
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by larcin** [comment]
[Authentication - JWT object is different between express and rails](https://trello.com/c/wfWMaGbv/3482-authentication-jwt-object-is-different-between-express-and-rails)

File: spec/requests/resources_spec.rb [hunk 1]
```diff
@@ -19,10 +19,20 @@
...
+    teams: 'Operations',
+    rendering_id: 16,
+    exp: Time.now.to_i + 2.weeks.to_i
+  }, ForestLiana.auth_secret, 'HS256')
+
   headers = {
     'Accept' => 'application/json',
     'Content-Type' => 'application/json',
-    'Authorization' => 'Bearer eyJhbGciOiJIUzI1NiJ9.eyJleHAiOiIxODQ5ODc4ODYzIiwiZGF0YSI6eyJpZCI6IjM4IiwidHlwZSI6InVzZXJzIiwiZGF0YSI6eyJlbWFpbCI6Im1pY2hhZWwua2Vsc29AdGhhdDcwLnNob3ciLCJmaXJzdF9uYW1lIjoiTWljaGFlbCIsImxhc3RfbmFtZSI6IktlbHNvIiwidGVhbXMiOiJPcGVyYXRpb25zIn0sInJlbGF0aW9uc2hpcHMiOnsicmVuZGVyaW5ncyI6eyJkYXRhIjpbeyJ0eXBlIjoicmVuZGVyaW5ncyIsImlkIjoxNn1dfX19fQ.U4Mxi0tq0Ce7y5FRXP47McNPRPhUx37LznQ5E3mJIp4'
+    'Authorization' => 'Bearer ' + token
```
Comment (arnaudbesnier) [inline spec/requests/resources_spec.rb]: [Code Style] You can do String interpolation in Ruby "Bearer #{token}".

File: spec/requests/resources_spec.rb [hunk 2]
```diff
@@ -19,10 +19,20 @@
     allow_any_instance_of(ForestLiana::PermissionsChecker).to receive(:is_authorized?) { true }
   end
 
+  token = JWT.encode({
+    id: 38,
+    email: 'michael.kelso@that70.show',
+    first_name: 'Michael',
+    last_name: 'Kelso',
+    teams: 'Operations',
```
Comment (arnaudbesnier) [inline spec/requests/resources_spec.rb]: [Spec] `teams` => `team`

File: app/services/forest_liana/login_handler.rb [hunk 3]
```diff
@@ -94,25 +94,13 @@ def is_two_factor_token_valid(user, two_factor_token)
...
-                id: rendering_id
-              }]
-            }
-          }
-        }
+        id: user['id'],
+        email: user['email'],
+        first_name: user['first_name'],
+        last_name: user['last_name'],
+        teams: user['teams'],
```
Comment (arnaudbesnier) [inline app/services/forest_liana/login_handler.rb]: [Spec]  `teams` => `team`

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # Change Log
 
 ## [Unreleased]
+### Changed
+- Technical - Makes the JWT lighter and consistent across lianas
```
Comment (arnaudbesnier) [inline CHANGELOG.md]: [Code Style]
```suggestion
- Technical - Makes the JWT lighter and consistent across lianas.
```

File: app/services/forest_liana/login_handler.rb (hunk 3 above)
Comment (arnaudbesnier) [reply app/services/forest_liana/login_handler.rb]: [Spec] `user['teams']` => `user['teams'][0]` I guess

File: app/controllers/forest_liana/application_controller.rb [hunk 5]
```diff
@@ -25,14 +25,10 @@ def self.papertrail?
...
       #         Forest with Papertrail.
       define_method :user_for_paper_trail do
-        forest_user['data']['data']['email']
+        @jwt_decoded_token['email']
       end
     end
 
-    def forest_user
-      @jwt_decoded_token
-    end
```
Comment (arnaudbesnier) [inline app/controllers/forest_liana/application_controller.rb]: [Regression] `forest_user` is a function used for customization, see this example (in Rails):
https://docs.forestadmin.com/documentation/reference-guide/routes#what-is-a-smart-route

So I am not sure we should delete this. We should add a comment to explain what is this method for by the way.

File: app/controllers/forest_liana/application_controller.rb [hunk 6]
```diff
@@ -69,7 +65,12 @@ def authenticate_user_from_jwt
 
           @jwt_decoded_token = JWT.decode(token, ForestLiana.auth_secret, true,
             { algorithm: 'HS256', leeway: 30 }).try(:first)
-          @rendering_id = @jwt_decoded_token['data']['relationships']['renderings']['data'][0]['id']
+
+          if @jwt_decoded_token['data']
+            raise ForestLiana::Errors::HTTP401Error.new("Deprecated token format")
```
Comment (arnaudbesnier) [inline app/controllers/forest_liana/application_controller.rb]: [Wording] Choose a consistent sentence (with express lianas) to explain why the user is logged out.
"Deprecated token format" is a bit cryptic.
"Your token format is invalid, please login again." is probably better, ask for help for the wording if you need.

**Pull request merged**